solutions/usaco-663.mdx

```diff
@@ -7,6 +7,105 @@ author: Maggie Liu, Dong Liu
 
 [Official Analysis (Java)](http://www.usaco.org/current/data/sol_square_bronze_dec16.html)
 
+## Video Solution
+
+By Maggie Liu
+
+<Youtube id="gXNTflZeutk" />
+
+<Spoiler title="Video Solution Code">
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <cstdio>
+using namespace std;
+
+int main()
+{
+	freopen("square.in", "r", stdin);
+	freopen("square.out", "w", stdout);
+	int x1, y1, x2, y2;
+	int x3, y3, x4, y4;
+	cin >> x1 >> y1 >> x2 >> y2 >> x3 >> y3 >> x4 >> y4;
+	int left, right, top, bottom;
+	
+	// find the sides of the smallest rectangle covering both pastures
+	left = min(x1, x3);
+	right = max(x2, x4);
+	bottom = min(y1, y3);
+	top = max(y2, y4);
+	
+	/*
+	 * the smallest square will need a side length
+	 * that is the maximum of the side lengths of the rectangle
+	 */
+	int side = max(right - left, top - bottom);
+	cout << side * side << endl;
+}
+```
+
+</CPPSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class SquarePasture
+{
+	public static void main(String[] args) throws IOException
+	{
+		Kattio io = new Kattio("square");
+		int x1 = io.nextInt(), y1 = io.nextInt(), x2 = io.nextInt(), y2 = io.nextInt();
+		int x3 = io.nextInt(), y3 = io.nextInt(), x4 = io.nextInt(), y4 = io.nextInt();
```

Comment:
80 chars?

Reply:
<img width="425" alt="image" src="https://user-images.githubusercontent.com/81596991/198904114-1221d0d7-26d4-444b-8fee-5a4c6d9e26c4.png">
79

Reply:
that doesn't include indentation

Reply:
okok